pkg/runtime/ports_test.go

```
@@ -0,0 +1,122 @@
+}
+
+func TestIsPortFree_Free(t *testing.T) {
+	t.Parallel()
+
+	// Get a free port, close it, then verify IsPortFree returns true
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatalf("Failed to get a free port: %v", err)
+	}
+	port := l.Addr().(*net.TCPAddr).Port
+	l.Close()
```

golangci-lint (2.12.1)

[error] Error return value of `l.Close` is not checked

(errcheck)